TypeScript/test/mocha/gilded-rose.spec.ts

```diff
@@ -5,6 +5,72 @@ describe('Gilded Rose', () => {
   it('should foo', () => {
     const gildedRose = new GildedRose([new Item('foo', 0, 0)]);
     const items = gildedRose.updateQuality();
-    expect(items[0].name).to.equal('fixme');
+    expect(items[0].name).to.equal('foo');
+  });
+  it('should degrade', () => {
+    const gildedRose = new GildedRose([new Item('Cheese', 5, 20)]);
+    const items = gildedRose.updateQuality();
+    expect(items[0].quality).to.equal(19);
+  });
+  it('should degrade twice as fast past the expiration date', () => {
+    const gildedRose = new GildedRose([new Item('Cheese', 0, 20)]);
+    const items = gildedRose.updateQuality();
+    expect(items[0].quality).to.equal(18);
+  });
+  it('should not have negative quality', () => {
+    const gildedRose = new GildedRose([new Item('test', 0, 0)]);
+    const items = gildedRose.updateQuality();
+    expect(items[0].quality).to.equal(0);
+  });
+  it('should increase quality for aged brie', () => {
+    const gildedRose = new GildedRose([new Item('Aged Brie', 2, 25)]);
+    const items = gildedRose.updateQuality();
+    expect(items[0].quality).to.equal(26);
+  });
+  it('should not increase quality past 50', () => {
+    const gildedRose = new GildedRose([new Item('Aged Brie', 0, 50)]);
+    const items = gildedRose.updateQuality();
+    expect(items[0].quality).to.equal(50);
+  });
+  it('should not decrease quality or sellIn value for Sulfuras', () => {
+    const gildedRose = new GildedRose([new Item('Sulfuras, Hand of Ragnaros', 30, 50)]);
+    const items = gildedRose.updateQuality();
+    expect(items[0].quality).to.equal(50);
+    expect(items[0].sellIn).to.equal(30);
+  });
+  it('should increase quality for backstage passes', () => {
```

Comment:
you could group the items specific tests by using nested describe

See https://stackoverflow.com/questions/19298118/what-is-the-role-of-describe-in-mocha